src/main/java/org/spongepowered/api/map/decoration/MapDecoration.java

```diff
@@ -0,0 +1,164 @@
+ * Valid coordinates range from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+ */
+public interface MapDecoration extends DataSerializable {
+
+    static Builder builder() {return Sponge.getRegistry().createBuilder(Builder.class);}
+
+    static MapDecoration of(MapDecorationType type, int x, int y, Direction rotation) {return builder().type(type).x(x).y(y).rotation(rotation).build();}
+
+    /**
+     * Gets the {@link MapDecorationType} of this MapDecoration
+     * @return MapDecorationType The type of the MapDecoration
+     */
+    MapDecorationType getType();
+
+    /**
+     * Gets the position of this MapDecoration, where it is right
+     * now, or where it would be if applied to a {@link org.spongepowered.api.map.MapInfo}.
+     * 0,0 is the centre of a map.
+     * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+     * @return Vector2i Co-ordinate position in world
+     */
+    Vector2i getPosition();
+
+    /**
+     * Gets the X position on a map, or where it will be when applied
+     * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+     * @return int Y world coordinate
+     */
+    int getX();
+
+    /**
+     * Gets the Y position on a map. 0 is the center of the map.
+     * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+     * @return int Y world coordinate
+     */
+    int getY();
+
+    /**
+     * Sets the position of where the MapDecoration is on Maps,
+     * or where it would be if applied to a {@link org.spongepowered.api.map.MapInfo}
+     * 0,0 is the center of the map
+     * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+     * @param position Vector2i world x and y cords
+     */
+    void setPosition(Vector2i position);
+
+    /**
+     * Sets the X position of where this MapDecoration is on a map
+     * or where it will be when applied
+     * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+     * @param x world cordinate
+     */
+    void setX(int x);
+
+    /**
+     * Sets the X position of where this MapDecoration is on a map.
+     * 0 is the center of the map.
+     * Ranges from {@value java.lang.Byte#MIN_VALUE}-{@value java.lang.Byte#MAX_VALUE}. AKA any valid byte value
+     * @param y pos
+     */
+    void setY(int y);
+
+    /**
+     * Sets rotation with a Direction. Must be a cardinal/ordinal/secondary ordinal
+     * @param rot Direction, a, secondary ordinal
+     */
+    void setRotation(Direction rot);
+
+    /**
+     * Gets the direction the Map Decoration is pointing
+     * @return Direction, a cardinal/ordinal/secondary
+     */
+    Direction getRotation();
+
+    interface Builder extends DataBuilder<MapDecoration> {
```

Comment:
This is really a sweeping comment here - please _do not_ have developer facing builders implement `DataBuilder`. They are two completely different concepts. I just looked at the implementation of this builder and there is _zero relation_ between the two functions. `build()` uses the methods in `Builder`, `DataBuilder` does not at all.

Keep them separate - that includes in the implementation. You don't even need to have an interface extend `DataBuilder` anywhere.